### DISCUSSION TITLE
Apple keychain support for creating account and login account

### DISCUSSION THREAD

**Discussion by rwzdoorn** [home-assistant/frontend#9923] [Other feature requests]
As a Mac user I'm using the Apple Keychain on my Mac and iOS devices. When signing up an account it would be convenient to  save the credentials in the Apple Keychain and auto-fill when logging in. 

Add the following to the create account screen:

<input id="new-password-text-field" type="password" **autocomplete="new-password"**/>
<input id="confirm-password-text-field" type="password" **autocomplete="new-password"**/>

Add the following to the change password screen:

<input id="user-text-field" type="email" **autocomplete="username"**/>
<input id="password-text-field" type="password" **autocomplete="current-password"**/>

https://developer.apple.com/documentation/security/password_autofill/enabling_password_autofill_on_an_html_input_element

**Comment by tathamoddie** [comment]
It looks like [these are standard values in HTML5 now](https://html.spec.whatwg.org/multipage/form-control-infrastructure.html#attr-fe-autocomplete), and [Chrome follows the same pattern](https://www.chromium.org/developers/design-documents/form-styles-that-chromium-understands) as per the Apple documentation link you provided.

Although it looks like this this is easier said than done.

The Home Assistant frontend uses [LitElements](https://lit.dev/).

For form inputs, it uses [iron-input](https://github.com/PolymerElements/iron-input). At least, it does so on the login page.

The actual `<input>` element lives in a shadow DOM.

Autocomplete doesn't work in this scenario, per https://github.com/lit/lit-element/issues/1023

There are underlying and unresolved bugs in every major browser engine:
* [Webkit bug #172567](https://bugs.webkit.org/show_bug.cgi?id=172567)
* [Chromium bug #649162](https://bugs.chromium.org/p/chromium/issues/detail?id=649162)
* [Firefox bug #1629226](https://bugzilla.mozilla.org/show_bug.cgi?id=1629226)

There are [potential workarounds](https://github.com/PolymerElements/iron-form/issues/197#issuecomment-579793392), but they're out of my current comfort zone to whip up a pull-request for.

**Reply by tathamoddie** [reply]
Further research: this is already being tracked in https://github.com/home-assistant/frontend/issues/3133.

It looks like some password manager focused fixes have just been landed recently in https://github.com/home-assistant/frontend/pull/9369, thanks to a broader accessibility push by @rianadon

**Reply by rianadon** [reply]
Thanks for the ping!

That PR already implemented a workaround for these browser bugs, so there shouldn't be any additional workarounds needed.

**Comment by rianadon** [comment]
If anyone wants to take a stab at implementing this, the best place would be inside the `render_input` method in `src/auth/ha-password-manager-polyfill.ts`.

These elements are placed outside the Shadow DOM, so the browser can see process these attributes (see the list of bugs in tahamoddie's answer).
